pkg/license/license.go

```diff
@@ -101,3 +101,31 @@ func Sync(a *app.App, licenseData string) (*kotsv1beta1.License, error) {
 
 	return latestLicense, nil
 }
+
+func GetCurrentLicense(a *app.App) (*kotsv1beta1.License, error) {
```

Comment:
Why? This feels like a dangerous way to hide the fact that archive is being fetched from S3 (we had this problem in the node version). This functionality in this file is simply a thin wrapper on kotsutil load. I think this file would be better removed and the caller should be responsible for fetching the archive  and getting the license from kotsutil.  In the end, we end up needing most of the kotskinds specs because they get pulled in for template rendering.

Reply:
removed this one, but I left the 'get license file from s3 tarball' helper

Reply:
@laverya :+1: how about renaming this function then to something like "DownloadAndExtractArchiveToGetLicense" (ok, but you get the point).

I think callers will start to assume this is cached in the database and making it more obvious that it's not feels like a way to avoid 1 rest endpoint that retrieves the entire bundle multiple times.